src/utils/test/latency_tracer_test.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include <gtest/gtest.h>
+#include <dsn/utility/latency_tracer.h>
+#include <dsn/dist/fmt_logging.h>
+
+using namespace dsn::utility;
+
+TEST(latency_tracer, add_point)
+{
+    int tracer1_stage_count = 3;
+    int tracer2_stage_count = 2;
+    int sub_tracer_stage_count = 2;
+
+    std::shared_ptr<latency_tracer> tracer1 =
+        std::make_shared<latency_tracer>(0, "stage0", "type1");
+    std::shared_ptr<latency_tracer> tracer2 =
+        std::make_shared<latency_tracer>(1, "stage0", "type1");
+    std::shared_ptr<latency_tracer> sub_tracer =
+        std::make_shared<latency_tracer>(2, "stage0", "link");
+
+    for (int i = 1; i < tracer1_stage_count; i++) {
```

Comment:
The loop is start with `1` which is a bit of strange because you add a point in constructor, as what I comment above, you don't have to add a point in c'tor.